grove/plugin/jwt_auth.go

```diff
@@ -0,0 +1,162 @@
+package plugin
+
+/*
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	log "github.com/apache/trafficcontrol/lib/go-log"
+	"github.com/dgrijalva/jwt-go"
+	"github.com/dgrijalva/jwt-go/request"
+	"github.com/iancoleman/strcase"
+)
+
+type JwtSecret struct {
+	Issuer   string `json:"iss"`
+	Audience string `json:"aud"`
+	Algoritm string `json:"alg"`
+	Kid      string `json:"kid"`
+	Value    string `json:"value"`
+}
+
+type JwtConfig struct {
+	ExportClaims []string    `json:"export_claims"`
+	Secrets      []JwtSecret `json:"secrets"`
+}
+
+type JwtContext struct {
+	SecretsIndex map[string]interface{}
+}
+
+func init() {
+	AddPlugin(10000, Funcs{onRequest: jwtAuth, startup: jwtStartup, load: jwtLoad})
+}
+
+func jwtAuth(icfg interface{}, d OnRequestData) bool {
+	if d.R.Method == http.MethodOptions {
+		return false
+	}
+	jwtContext := (*d.Context).(JwtContext)
+	jwtConfig := icfg.(*JwtConfig)
```

Comment:
Do you mind adding a nil check and return, if this is `nil`?
```
if icfg == nil {
  return
}
```
That will let Grove run instead of crashing, if the config is nil.